### PR TITLE
feat: add MarketSummary query

### PR DESCRIPTION
**Pull request by toteki**
## Description

Adds a query which groups a set of queries the frontend often executes together.

Needs discussion:
- Does frontend `exchange_rate` requested in [client #434](https://github.com/umee-network/umee-v1-client/issues/434) refer to asset's USD price (`x/oracle` query) or its uToken exchange rate (`x/leverage` query)? Currently returns both.
- To avoid mixing _usd totals_ and _token totals_ in the query response, would it be acceptable to return token market size (e.g. `5M ATOM` instead of usd market size (e.g. `100M USD`)? Frontend would then multiple by oracle price.

closes: #893

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/umee/leverage/v1/query.proto [hunk 1]
```diff
@@ -356,3 +361,30 @@ message QueryLiquidationTargetsRequest {}
...
+  string denom = 1;
+}
+
+// QueryMarketSummaryResponse defines the response structure for the
+// MarketSummary gRPC service handler.
+message QueryMarketSummaryResponse {
+  string symbol_denom = 1;
+  uint32 exponent = 2;
+  string oracle_price = 3
+  [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = true];
```
Comment (toteki) [inline proto/umee/leverage/v1/query.proto:377]: Note how nullable is true here - if oracle is down, this field will be null but the query will work.

File: proto/umee/leverage/v1/query.proto [hunk 2]
```diff
@@ -356,3 +361,30 @@ message QueryLiquidationTargetsRequest {}
...
+  string uToken_exchange_rate = 4
+  [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string lend_APY = 5
+  [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string borrow_APY = 6
+  [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec", (gogoproto.nullable) = false];
+  string market_size = 7
+  [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", (gogoproto.nullable) = false];
```
Comment (toteki) [inline proto/umee/leverage/v1/query.proto:385]: Note `market_size` (tokens) not `market_size_usd` here. See discussion question in PR description

**Comment by EbrahimUmee** [comment]
`Does frontend exchange_rate requested in https://github.com/umee-network/umee-v1-client/issues/434 refer to asset's USD price (x/oracle query) or its uToken exchange rate (x/leverage query)? Currently returns both.`

This is the assets USD price (Double check)

`To avoid mixing usd totals and token totals in the query response, would it be acceptable to return token market size (e.g. 5M ATOM instead of usd market size (e.g. 100M USD)? Frontend would then multiple by oracle price`

Yes if you believe thats a necessary change we can do that.

**Comment by toteki** [comment]
It's correct - the 10^6 difference is between `uatom` and `ATOM` because `exponent` = 6

**Review by RafilxTenfen — APPROVED** [review]

**Review by adamewozniak — APPROVED** [review]

**Pull request merged**